### PR TITLE
[Editorial] - Euclidean Minimum Spanning Tree - Part 1

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by brebenelmihnea** [comment]
You can directly make commits

File: content/5_Plat/Geo_Pri.mdx [hunk 1]
```diff
@@ -384,6 +384,208 @@ int main() {
...
+#include <vector>
+
+using namespace std;
+
+struct point {
+	int x, y, idx;
+};
+
+// BeginCodeSnip{DSU}
+class DisjointSets {
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:412]: amazing!

File: content/5_Plat/Geo_Pri.mdx [hunk 2]
```diff
@@ -384,6 +384,208 @@ int main() {
...
+	mst = compute_mst(mst);
+	vector<vector<pair<int, int>>> tree(n);
+	for (pair<point, point> edge : mst) {
+		tree[edge.first.idx].push_back(
+		    {edge.second.idx,
+		     dist(points[edge.first.idx], points[edge.second.idx])});
+		tree[edge.second.idx].push_back(
+		    {edge.first.idx,
+		     dist(points[edge.first.idx], points[edge.second.idx])});
+		// cout << edge.first.idx << ' ' << edge.second.idx << '\n';
```
Comment (ryanchou-dev) [inline content/5_Plat/Geo_Pri.mdx:559]:
```suggestion
```

File: content/5_Plat/Geo_Pri.mdx [hunk 3]
```diff
@@ -384,6 +384,208 @@ int main() {
...
+
+```cpp
+#include <functional>
+#include <iostream>
+#include <unordered_set>
+#include <vector>
+
+using namespace std;
+
+struct point {
```
Comment (ryanchou-dev) [inline content/5_Plat/Geo_Pri.mdx:407]: capitalize structs?

**Review by ryanchou-dev — APPROVED** [review]
:D

**Comment by brebenelmihnea** [comment]
@SansPapyrus683 why not approve :))

File: content/5_Plat/Geo_Pri.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:559]: Yep

File: content/5_Plat/Geo_Pri.mdx [hunk 4]
```diff
@@ -384,6 +384,207 @@ int main() {
...
 </LanguageSection>
 
+### Euclidean minimum spanning tree - EMST
+
+<FocusProblem problem="emst"/>
+
+#### Initial Approach
+
+The problem is a combination of EMST - euclidean minimum spanning tree - and LCA. We can compute the EMST using Kruskal's or Prim's algorithm. Using this approach, the algorithm
+will run on $n^2$ edges, which causes TLE.
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:394]: > Using this approach, the algorithm will run on $n^2$ edges, which causes TLE.

so does this sol work

Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:394]: It works, but id gets TLE

Comment (SansPapyrus683) [reply content/5_Plat/Geo_Pri.mdx:394]: uh then what's the point of this solution

Comment (SansPapyrus683) [reply content/5_Plat/Geo_Pri.mdx:394]: OH I JUST SAW

File: content/5_Plat/Geo_Pri.mdx [hunk 5]
```diff
@@ -384,6 +384,213 @@ int main() {
...
+		for (int i = MAX_LOG; i >= 0; i--) {
+			if (up[x][i] != up[y][i]) {
+				x = up[x][i];
+				y = up[y][i];
+			}
+		}
+		return up[x][1];
+	}
+
+	int query(int x, int y) {
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:497]: doc comemnt or more descriptive name

Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx:498]: Done

File: content/5_Plat/Geo_Pri.mdx [hunk 6]
```diff
@@ -384,6 +384,213 @@ int main() {
...
+	return mst;
+}
+
+int main() {
+	int test_num;
+	cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		int n;
+		cin >> n;
+		vector<Point> Points;
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:549]: rename to `points`

Comment (brebenelmihnea) [reply content/5_Plat/Geo_Pri.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]
1. proof of tle
2. check that my new comments actually make sense

**Comment by brebenelmihnea** [comment]
@SansPapyrus683 
https://www.codechef.com/viewsolution/1060700768

**Comment by SansPapyrus683** [comment]
image

hm idt this is tle lol

**Comment by brebenelmihnea** [comment]
?

**Comment by SansPapyrus683** [comment]
1. you say "oh this is too slow"
2. usually, algorithms that are too slow result in tle
3. the screenshot says rte

File: content/5_Plat/Geo_Pri.mdx [hunk 7]
```diff
@@ -384,6 +385,234 @@ int main() {
...
+}
+```
+
+</CPPSection>
+</LanguageSection>
+
+#### Solution
+
+To improve the previous discussed algorithm we'll reduce the complete graph - with $N^2$ edges - to a much smaller one; this way the $\mathcal{O}(N \codt \log{N})$ Kruskal algorithms will
+run on fewer edges, thus speeding up the computation. There several ways of reducing the number of edges; two of them are:
```
Comment (SansPapyrus683) [inline content/5_Plat/Geo_Pri.mdx:593]: would prefer you made this in a diff pr but ok

**Comment by SansPapyrus683** [comment]
> 1. you say "oh this is too slow"
> 2. usually, algorithms that are too slow result in tle
> 3. the screenshot says rte

btw wb this

**Pull request closed without merging**